.github/workflows/ci.yml

```diff
@@ -62,15 +62,39 @@ jobs:
       #   run: pnpm --filter desktop run coverage
       - name: lint
         run: tea -E pnpm -r lint
-      - name: chec
+      - name: check
         run: tea -E pnpm -r check
+
+  e2e_test:
+    needs: changes
+    runs-on: macos-latest
+    env:
+      # see: https://github.com/electron-userland/electron-builder/issues/3179
+      USE_HARD_LINKS: false
+    steps:
+      - uses: teaxyz/setup@v0
+        with:
+          version: 0.31.2
+      - uses: actions/checkout@v3
+      - name: env file
+        run: cp modules/desktop/.env.example modules/desktop/.env
+      - name: install app dependencies
+        run: tea -E xc setup
+      - name: test build electron main process
+        run: tea -E xc build
+        env:
+          CSC_IDENTITY_AUTO_DISCOVERY: "false"
+          MAC_BUILD_TARGET: "dir"
+      - name: e2e test
```

Comment:
we probably need to run this again on build_sign_notarize.yml after build-dev or build-release just to be sure